### PR TITLE
Dc 4.13

### PR DESCRIPTION
**Pull request by carlocaione**
https://phabricator.endlessm.com/T11531

### CODE REVIEW

File: drivers/dma-buf/reservation.c [hunk 1]
```diff
@@ -195,8 +195,7 @@ reservation_object_add_shared_replace(struct reservation_object *obj,
 	if (old)
 		kfree_rcu(old, rcu);
 
-	if (old_fence)
```
Comment (dsd) [inline drivers/dma-buf/reservation.c]: In the commit message please mention the commit ID of the staging branch that you got this from, and the exact commands you used to take the changes from that branch and apply them here.

File: drivers/gpu/drm/amd/display/dc/dce110/dce110_resource.c [hunk 2]
```diff
@@ -891,14 +891,21 @@ static bool dce110_validate_surface_sets(
 		if (context->stream_status[i].plane_count > 2)
 			return false;
 
-		if ((context->stream_status[i].plane_states[i]->format >= SURFACE_PIXEL_FORMAT_VIDEO_BEGIN) &&
```
Comment (dsd) [inline drivers/gpu/drm/amd/display/dc/dce110/dce110_resource.c]: please add a commit message with the basic reasoning/links here

File: drivers/gpu/drm/amd/display/dc/core/dc_resource.c [hunk 3]
```diff
@@ -1416,7 +1416,7 @@ static struct audio *find_first_free_audio(
 {
 	int i;
 	for (i = 0; i < pool->audio_count; i++) {
-		if ((res_ctx->is_audio_acquired[i] == false) && (res_ctx->is_stream_enc_acquired[i] == true)) {
```
Comment (dsd) [inline drivers/gpu/drm/amd/display/dc/core/dc_resource.c]: please add a commit message with the basic reasoning/links here

File: drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c [hunk 4]
```diff
@@ -36,6 +36,7 @@
 
 #include <drm/drm_pciids.h>
 #include <linux/console.h>
+#include <linux/dmi.h>
```
Comment (dsd) [inline drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c]: in the commit message please briefly mention which tests you have done on this unit

**Comment by dsd** [comment]
I think you also need a commit to turn this on in the config files under debian.master (you can see how I did this on the master branch)

**Comment by carlocaione** [comment]
@dsd PR respin ready

**Comment by dsd** [comment]
Looks good. Must not be merged before eos3.3 is open for 2018-01 release.

**Comment by jprvita** [comment]
This will break the build on OBS because the following switches are not defined: `SND_SOC_AMD_CZ_RT5645_MACH` and `SND_SOC_AMD_ACP3x`.

I use the following script to check these things (redirecting to `.config`) followed by `make silentoldconfig`, which needs to finish without requiring user input:

```
#!/bin/bash
SOURCE_FILES="debian.master/config/config.common.ubuntu \
              debian.master/config/amd64/config.common.amd64 \
              debian.master/config/amd64/config.flavour.generic"
cat $SOURCE_FILES | sed -e 's/.*CONFIG_VERSION_SIGNATURE.*/CONFIG_VERSION_SIGNATURE=""/'
```
I should probably make it a little bit more automatic and commit this to the repo, maybe running it inside some git hook.

**Comment by carlocaione** [comment]
@jprvita thanks for noticing. New PR ready.

**Comment by dsd** [comment]
merged

**Pull request closed without merging**